cmd/agent/main.go

```diff
@@ -95,6 +122,50 @@ func sockDial(serverAddr string, socksProxy string, socksUser string, socksPass
 	return proxyDialer.Dial("tcp", serverAddr)
 }
 
+func wsconnect(config *tls.Config, wsaddr string, proxystr string, useragent string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+	defer cancel()
+
+	//proxystr = "http://admin:secret@127.0.0.1:8080"
+	proxyUrl, err := url.Parse(proxystr)
+	if err != nil || proxystr == "" {
+		proxyUrl = nil
+	}
+	httpTransport := &http.Transport{
+		MaxIdleConns:    http.DefaultMaxIdleConnsPerHost,
+		TLSClientConfig: config,
+		Proxy:           http.ProxyURL(proxyUrl),
+	}
+
+	httpClient := &http.Client{Transport: httpTransport}
+
+	httpheader := &http.Header{}
+	httpheader.Add("User-Agent", useragent)
+
+	wsConn, _, err := websocket.Dial(ctx, wsaddr, &websocket.DialOptions{HTTPClient: httpClient, HTTPHeader: *httpheader})
+	if err != nil {
+		return err
+	}
+
+	netctx, cancel := context.WithTimeout(context.Background(), time.Hour*999999)
```

Comment:
Why `time.Hour*999999`? Why not use a meaningful timeout or just `context.Background()`?

Reply:
It means endless connection. I do not remember exactly why 99999 (I think it from my early experiments with yamux+websocket). In this case of cause it can be replaced by context.Background()
`netConn := websocket.NetConn(context.Background(), wsConn, websocket.MessageBinary)`
I tested It - works well.